src/search.c

```diff
@@ -153,7 +150,78 @@ static void free_bundle_result_data(void *data)
 	free(br);
 }
 
-static void print_final_results(void)
+/* unmark all bundle_results in bl as seen */
+static void unsee_bundle_results(struct list *bl)
+{
+	struct bundle_result *br;
+	struct list *ptr = list_head(bl);
+
+	while (ptr) {
+		br = ptr->data;
+		br->seen = false;
+		ptr = ptr->next;
+	}
+}
+
+/* recursively calculate size from a complete list of bundle_result structs */
+static long calculate_size(char *bname, struct list *bundle_info, bool installed)
+{
+	struct list *ptr;
+	struct bundle_result *bi;
+	long size = 0;
+	ptr = list_head(bundle_info);
+	while (ptr) {
+		bi = ptr->data;
+		ptr = ptr->next;
+		if (bi->seen || strcmp(bname, bi->bundle_name) != 0) {
+			continue;
+		}
+
+		/* this was a top-level match or an include match
+		 * add the contentsize if we are computing for an already-installed bundle
+		 * or if the bundle is not already included */
+		if (installed || !is_tracked_bundle(bname)) {
```

Comment:
What is installed signify here? I expect installed if and only if is_tracked_bundle but clearly that is not the case.

Reply:
@icarus-sparry 

`installed == true` indicates the initial call to this recursive function was for an installed bundle. In this case we want to calculate the contentsize of all included bundles (we know they are all installed already) so we can report the total installed size on the system. I did it this way because of your use case for running swupd search to see what the disk hogs on your system are. I can update the comment to explain that.